### PR TITLE
update: use label as selector for namespaces in cache

### PR DESCRIPTION
**Pull request by zdtsw**
## Description

- for clean installation

1.cache default hardcoded namespace's deployment and secrets
2. cache user created namespace as long as with label `opendatahub.io/watched-namespace=true` 's deployment and secrets
in this case, if user want to use their own namespace e.g `redhat-ods-application-2`, 
they would need to create this namespace before install ODH, and add this label in the  namespace.
then they can create DSCI and set applicationNamespace to `redhat-ods-application-2`
DSCI will add `opendatahub.io/generated-namespace=true` to this namespace `redhat-ods-application-2`
this helps if operator restart or later upgrade to new version, cache will still work. 

- for upgrade:
cache namespace with label `opendatahub.io/generated-namespace=true` 's deployment and secrets

## Merge criteria

- [ ] You have read the [contributors guide](https://github.com/opendatahub-io/opendatahub-operator/blob/incubation/CONTRIBUTING.md).
- [ ] Commit messages are meaningful - have a clear and concise summary and detailed explanation of what was changed and why.
- [ ] Pull Request contains a description of the solution, a link to the JIRA issue, and to any dependent or related Pull Request.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -206,8 +205,11 @@ func main() { //nolint:funlen,maintidx
 			&configv1.Authentication{}: {
 				Field: fields.Set{"metadata.name": cluster.ClusterAuthenticationObj}.AsSelector(),
 			},
-			// for prometheus and black-box deployment and ones we owns
-			&appsv1.Deployment{}: {Namespaces: deploymentCache},
+			// for deployments in application namespace + operator namespace + monitoring namespace + rhods-notebooks
+			// default wb namespace is not needed since no deployment only sfs,  but our label is on it so we cache it as well
+			&appsv1.Deployment{}: {
+				Label: labels.Set{"opendatahub.io/generated-namespace": "true"}.AsSelector(),
```
Comment (lburgazzoli) [inline main.go:211]: is "generated-namespace" added to a deployment ?

Comment (zdtsw) [reply main.go]: i messed up deployment and namespace

File: main.go [hunk 2]
```diff
@@ -351,40 +352,43 @@ func main() { //nolint:funlen,maintidx
...
-		namespaceConfigs["opendatahub"] = cache.Config{}
+	// TODO: if we want to not harcoded above two namespace we can add them with label selector
+	// maistra.io/member-of=istio-system
+	// network.openshift.io/policy-group=ingress
+
+	labelSelector := client.MatchingLabels{
+		"opendatahub.io/generated-namespace": "true",
+	}
+	namespaceList := &corev1.NamespaceList{}
+	if err := cli.List(ctx, namespaceList, labelSelector); err != nil {
```
Comment (lburgazzoli) [inline main.go:368]: on a fresh install, is this going to work ? my understading is that the application namespace is created by the DSCI controller so at the time this `createSecretCacheConfig`  is invoked, then the namespace my not be there yet

File: main.go [hunk 3]
```diff
@@ -351,40 +352,43 @@ func main() { //nolint:funlen,maintidx
...
-		namespaceConfigs["redhat-ods-applications"] = cache.Config{}
-	case cluster.SelfManagedRhoai:
-		namespaceConfigs["redhat-ods-applications"] = cache.Config{}
-	default:
-		namespaceConfigs["opendatahub"] = cache.Config{}
+	labelSelector := client.MatchingLabels{
+		"opendatahub.io/generated-namespace": "true",
+	}
+	namespaceList := &corev1.NamespaceList{}
+	if err := cli.List(ctx, namespaceList, labelSelector); err != nil {
```
Comment (lburgazzoli) [inline main.go:385]: probably the same as before

File: main.go (hunk 2 above)
Comment (zdtsw) [reply main.go:368]: you are right, this only works for upgrade case, 
clean installation wont get these namespaces

**Pull request closed without merging**